core/chains/evm/client/errors.go

```diff
@@ -161,6 +162,21 @@ var arbitrum = ClientErrors{
 	ServiceUnavailable:    regexp.MustCompile(`(: |^)502 Bad Gateway: [\s\S]*$`),
 }
 
+// Treasure
+// Derived from Arbitrum config due to similar behaviour
+var treasureFatal = regexp.MustCompile(`(: |^)(invalid message format|forbidden sender address)$|(: |^)(execution reverted)(:|$)`)
+var treasure = ClientErrors{
+	NonceTooLow:           regexp.MustCompile(`(: |^)invalid transaction nonce$|(: |^)nonce too low(:|$)`),
+	NonceTooHigh:          regexp.MustCompile(`(: |^)nonce too high(:|$)`),
+	TerminallyUnderpriced: regexp.MustCompile(`(: |^)gas price too low$`),
+	InsufficientEth:       regexp.MustCompile(`(: |^)(not enough funds for gas|insufficient funds for gas \* price \+ value)`),
+	Fatal:                 treasureFatal,
+	L2FeeTooLow:           regexp.MustCompile(`(: |^)max fee per gas less than block base fee(:|$)`),
+	L2Full:                regexp.MustCompile(`(: |^)(queue full|sequencer pending tx pool full, please try again)(:|$)`),
+	ServiceUnavailable:    regexp.MustCompile(`(: |^)502 Bad Gateway: [\s\S]*$`),
+	InvalidSender:         regexp.MustCompile(`(: |^)invalid chain id for signer(:|$)`),
+}
```

Comment:
You only have to include the unique errors for Treasure. There isn't a need to duplicate the Arbitrum errors here. Our error classification looks across all of the clients for matching errors.